src/openrct2/scripting/bindings/entity/ScGuest.cpp

```diff
@@ -494,6 +499,298 @@ namespace OpenRCT2::Scripting
         return DukValue::take_from_stack(ctx, -1);
     }
 
+    DukValue ScGuest::items_get() const
+    {
+        auto ctx = GetContext()->GetScriptEngine().GetContext();
+
+        duk_push_array(ctx);
+
+        auto peep = GetGuest();
+        if (peep != nullptr)
+        {
+            duk_uarridx_t index = 0;
+            for (const auto& itemEnumPair : ShopItemMap)
+            {
+                auto shopItem = itemEnumPair.second;
+                if (!peep->HasItem(shopItem))
+                {
+                    continue;
+                }
+
+                // GuestItem
+                auto obj = OpenRCT2::Scripting::DukObject(ctx);
+                obj.Set("type", itemEnumPair.first);
+
+                if (shopItem == ShopItem::Voucher)
+                {
+                    // Voucher
+                    obj.Set("voucherType", VoucherTypeMap[peep->VoucherType]);
+                    if (peep->VoucherType == VOUCHER_TYPE_RIDE_FREE)
+                    {
+                        // RideVoucher
+                        obj.Set("rideId", peep->VoucherRideId.ToUnderlying());
+                    }
+                    if (peep->VoucherType == VOUCHER_TYPE_FOOD_OR_DRINK_FREE)
+                    {
+                        // FoodDrinkVoucher
+                        obj.Set("item", ShopItemMap[peep->VoucherShopItem]);
+                    }
```

Comment:
They can't happen at the same time, right? So an `if/else` would suit best here.
```suggestion
                    if (peep->VoucherType == VOUCHER_TYPE_RIDE_FREE)
                    {
                        // RideVoucher
                        obj.Set("rideId", peep->VoucherRideId.ToUnderlying());
                    }
                    else if (peep->VoucherType == VOUCHER_TYPE_FOOD_OR_DRINK_FREE)
                    {
                        // FoodDrinkVoucher
                        obj.Set("item", ShopItemMap[peep->VoucherShopItem]);
                    }
```